R/input-check-helpers.R

```diff
@@ -0,0 +1,207 @@
+#' @title Check Prediction Input For Lower-level Scoring Functions
+#'
+#' @description
+#' Helper function to check inputs for lower-level score functions.
+#' @param predictions an object with predictions. Depending on whether
+#' `class = vector` or `class = "matrix"` this can be either a vector of length
+#' n (corresponding to the length of the true_values) or a nxN matrix of
+#' predictive samples, n (number of rows) being the number of data points and
+#' N (number of columns) the number of Monte Carlo samples
+#' @param type character, one of "continuous", "integer" or "binary" that
+#' defines the type of the forecast
+#' @param class character, either "vector" or "matrix" that determines the
+#' class the input has to correspond to
+#' @inheritParams ae_median_sample
+#' @return NULL
+#' @keywords internal
+
+check_predictions <- function(predictions,
+                              true_values = NULL,
+                              type = c("continuous", "integer", "binary"),
+                              class = c("vector", "matrix")) {
+  type <- match.arg(type)
+  class <- match.arg(class)
+
+  if (missing(predictions)) {
+    stop("argument 'predictions' missing")
+  }
+
+  if (class == "vector") {
+    if (!is.vector(predictions)) {
+      msg <- sprintf(
+        "'predictions' should be a vector. Instead `%s` was found",
+        class(predictions[1])
+      )
+      stop(msg)
+    }
+    if (!is.null(true_values) && length(predictions) != length(true_values)) {
+      msg <- sprintf(
+        "Mismatch: 'true_values' has length `%s`, but 'predictions' has length `%s`.", # nolint
+        length(true_values), length(predictions)
+      )
+      stop(msg)
+    }
+  }
+
+  if (class == "matrix") {
+    if (!is.matrix(predictions)) {
+      msg <- sprintf(
+        "'predictions' should be a matrix. Instead `%s` was found",
+        class(predictions[1])
+      )
+      stop(msg)
+    }
+    if (!is.null(true_values) && nrow(predictions) != length(true_values)) {
+      msg <- sprintf(
+        "Mismatch: 'true_values' has length `%s`, but 'predictions' has `%s` rows.",
+        length(true_values), nrow(predictions)
+      )
+      stop(msg)
+    }
+  }
+
+  if (type == "integer") {
+    if (isFALSE(all.equal(predictions, as.integer(predictions)))) {
```

Comment:
Please note that `isFALSE()` was introduced in R 3.5.0 and therefore increase the required R version number. I don't think it's a huge issue since R 3.5.0 has been out for quite some time now but it's something to keep in mind and `DESCRIPTION` should be updated accordingly.

Reply:
interesting that this didn't show up in any of the checks... I'll update the the DESCRIPTION. Thanks!